### PR TITLE
[Keyboard] Add greatsword80

### PR DESCRIPTION
**Pull request by EasonQian1**
## Description
Add greatsword80

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/eason/greatsword80/config.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+ */
```
Comment (waffle87) [inline keyboards/eason/greatsword80/config.h:34]:
```suggestion
```

File: keyboards/eason/greatsword80/greatsword80.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/* Copyright 2023 Eason <xiaoxuxk@yeah.net>
```
Comment (waffle87) [inline keyboards/eason/greatsword80/greatsword80.h:1]: This file should be removed, assuming the `matrix` data in `info.json` file is correct.

File: keyboards/eason/greatsword80/greatsword80.c [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2023 Eason <xiaoxuxk@yeah.net>
```
Comment (waffle87) [inline keyboards/eason/greatsword80/greatsword80.c:1]: This file can also be removed if above comment is true.

File: keyboards/eason/greatsword80/rules.mk [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (waffle87) [inline keyboards/eason/greatsword80/rules.mk:6]:
```suggestion
```

File: keyboards/eason/greatsword80/info.json [hunk 5]
```diff
@@ -0,0 +1,105 @@
+{
+    "keyboard_name": "GREATSWORD80",
+    "manufacturer": "Eason",
+    "url": "https://github.com/EasonQian1",
+    "maintainer": "Eason",
+    "usb": {
+        "vid": "0x980E",
+        "pid": "0x998F",
+        "device_version": "0.0.1"
+    },
```
Comment (waffle87) [inline keyboards/eason/greatsword80/info.json:10]:
```suggestion
    },
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
```

File: keyboards/eason/greatsword80/readme.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# GREATSWORD80
+A customizable soldering 80% keyboard.
+
+* Keyboard Maintainer: [eason](https://github.com/EasonQian1)
+* Hardware Supported: greatsword80
+* Hardware Availability: [eason](https://github.com/EasonQian1)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make eason/greatsword80:default
```
Comment (waffle87) [inline keyboards/eason/greatsword80/readme.md:10]:
```suggestion
    make eason/greatsword80:default
    
Flashing example for this keyboard:

    make eason/greatsword80:default:flash

```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
Please see the changes that I marked as unresolved.

**Comment by EasonQian1** [comment]
> Please see the changes that I marked as unresolved.
Hello buddy. I modified

**Comment by waffle87** [comment]
That's not what I meant. In `info.json` file, you have `matrix` coordinates:
eg.
```{"label": "K00 (B0,B6)", "matrix": [0, 0], "x": 0, "y": 0},```
This eliminates the need to define the layout macro in `greatsword.h` file.
So if the `matrix` data in `info.json` file is correct, `greatsword.c` & `greatsword.h` files should both be deleted.

**Comment by EasonQian1** [comment]
> That's not what I meant. In `info.json` file, you have `matrix` coordinates: eg. `{"label": "K00 (B0,B6)", "matrix": [0, 0], "x": 0, "y": 0},` This eliminates the need to define the layout macro in `greatsword.h` file. So if the `matrix` data in `info.json` file is correct, `greatsword.c` & `greatsword.h` files should both be deleted.

LOL！I see, I misunderstood. Thanks

**Comment by EasonQian1** [comment]
@waffle87 Hi buddy. Please help me to check if it is correct

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**